### PR TITLE
Generate and deploy reference assembly for Microsoft.PowerShell.Commands.Utility.dll

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Update `New-ReferenceAssembly` and `New-UnifiedNugetPackage` to generate reference assembly for `Microsoft.PowerShell.Commands.Utility.dll` and properly deploy it for `Microsoft.PowerShell.Commands.Utility` NuGet package and `Microsoft.PowerShell.SDK` NuGet package.

An incremental step to fix, eventually, #8121

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -1861,23 +1884,78 @@ function New-ReferenceAssembly
...
+    $assemblyNames = @(
+        "System.Management.Automation",
+        "Microsoft.PowerShell.Commands.Utility"
+    )
+
+    foreach ($assemblyName in $assemblyNames) {
+
+        $projectFolder = New-Item -Path "$genAPIFolder/$assemblyName" -ItemType Directory -Force
+        $generatedSource = Join-Path $projectFolder "$assemblyName.cs"
+        $filteredSource = Join-Path $projectFolder "${assembly}_Filtered.cs"
```
Comment (PaulHigin) [inline tools/packaging/packaging.psm1]: "${assemblyName}_Filtered.cs" ?

**Review by PaulHigin — DISMISSED** [review]

File: tools/packaging/packaging.strings.psd1 [hunk 2]
```diff
@@ -148,7 +155,30 @@ RefAssemblyCsProj = @'
...
+    <SignAssembly>true</SignAssembly>
+    <LangVersion>Latest</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="System.Management.Automation">
+    <HintPath>{2}</HintPath>
+    </Reference>
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.strings.psd1]: How do we manage updating versions of these references when an update is published for them?
Can you open an issue to track this work?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: tools/packaging/packaging.strings.psd1 (hunk 2 above)
Comment (daxian-dbw) [reply tools/packaging/packaging.strings.psd1]: Yes, issue opened: #8721

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -1698,6 +1696,40 @@ function New-UnifiedNugetPackage
     }
 }
 
+function CopyReferenceAssemblies
```
[the remark this answers is not in the document]
Comment (daxian-dbw) [reply tools/packaging/packaging.psm1]: Done.

File: tools/packaging/packaging.strings.psd1 [hunk 4]
```diff
@@ -132,14 +137,16 @@ NuspecTemplate = @'
...
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <Version>{0}</Version>
     <DelaySign>true</DelaySign>
     <AssemblyOriginatorKeyFile>{1}</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
+    <LangVersion>Latest</LangVersion>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.Management.Infrastructure" Version="1.0.0-alpha08" />
```
[the remark this answers is not in the document]
Comment (daxian-dbw) [reply tools/packaging/packaging.strings.psd1]: Done.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@TravisEz13 @PaulHigin Please have another look.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**